contrib/net/http/http.go

```diff
@@ -65,6 +67,19 @@ func (mux *ServeMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+// patternRoute returns the route part of a go1.22 style ServeMux pattern. I.e.
+// it returns "/foo" for the pattern "/foo" as well as the pattern "GET /foo".
+func patternRoute(s string) string {
```

Comment:
We also need to report the span tag `http.route` - this key used by all Datadog API Catalogs which doesn't work if http.route is not set.

But it should be the full route pattern, such as `http.route: /user/{uid}`, in the string format used by the framework.
I don't know how to get this information with go1.22.

Reply:
I was under the impression that my patch handles this?

![2024-05-30 contribnethttp Support go1 22 ServeMux patterns by felixge · Pull Request #2716 · DataDogdd-trace-go at 09 56 22@2x](https://github.com/DataDog/dd-trace-go/assets/15000/269d0a91-43e3-4f9d-87e0-44254227f317)